File: app/src/main/java/com/song/sunset/utils/preinstall/PreinstallHandler.java
package com.song.sunset.utils.preinstall;

/**
 * @author songmingwen
 * @description
 * @since 2018/10/9
 */
public interface PreinstallHandler {

    String getPreinstallInfo();

    void setNextHandler(PreinstallHandler preinstallHandler);
}

File: app/src/main/java/com/song/sunset/utils/preinstall/PreinstallHandler.kt
package com.song.sunset.utils.preinstall

/**
 * @author songmingwen
 * @description
 * @since 2018/10/9
 */
interface PreinstallHandler {

    val preinstallInfo: String

    fun setNextHandler(preinstallHandler: PreinstallHandler)
}
